File: Chapter_1/namenum/namenum.java

/*
ID: 100021881
LANG: JAVA
PROG: namenum
*/

import java.io.*;
import java.util.*;

public class namenum {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new FileReader("namenum.in"));
        BufferedReader dic = new BufferedReader(new FileReader("dict.txt"));
        PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter("namenum.out")));
        long tag = Long.parseLong(br.readLine());
        String number = String.valueOf(tag);
        List<String> touchtone = new ArrayList<String>();
        touchtone.add("ABC");
        touchtone.add("DEF");
        touchtone.add("GHI");
        touchtone.add("JKL");
        touchtone.add("MNO");
        touchtone.add("PRS");
        touchtone.add("TUV");
        touchtone.add("WXY");

        List<Integer> digits = new ArrayList<Integer>();
        List<String> names = new ArrayList<String>();
        List<String> temp = new ArrayList<String>();
        for (int i = 0; i < number.length(); i++) {
            digits.add(Integer.parseInt(String.valueOf(number.charAt(i))));
        }
        for (int i : digits) {
            temp.add(touchtone.get(i - 2));
        }
        String line;
        while ((line = dic.readLine()) != null) {
            boolean same = true;
            if (temp.size() == line.length()) {
                for (int i = 0; i < temp.size(); i++) {
                    if (temp.get(i).indexOf(line.charAt(i)) == -1) {
                        same = false;
                        break;
                    }
                }
                if (same)
                    names.add(line);
            }
        }
        if (names.size() > 0) {
            for (String name : names) {
                out.println(name);
            }
        } else {
            out.println("NONE");
        }
        dic.close();
        out.close();
        br.close();
    }
}

File: Chapter_1/namenum/namenum.cpp
/*
 ID: 100021881
 LANG: C++
 PROG: namenum
*/
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

int main() {
    ofstream out("namenum.out");
    ifstream in("namenum.in");
    ifstream dict("dict.txt");
    string id;
    in >> id;
    bool found = false;
    string name;
    while (dict >> name) {
        if (name.length() == id.length()) {
            string number = "";
            for (int i = 0; i < name.length(); i++) {
                if (name[i] == 'Q' || name[i] == 'Z') goto label;
                if (name[i] < 'Q') number += ((name[i]-'A')/3) + '2';
                else number += ((name[i]-'Q')/3)+'7';
            }
            if (number!=id) label:continue;
            found = true;
            out << name << endl;
        }
    }
    if (!found) out << "NONE" << endl;
}